mteb/models/clip_models.py

```diff
@@ -0,0 +1,60 @@
+from PIL import Image
+from transformers import AutoProcessor, AutoModel
+from typing import Any
+import torch
+from tqdm import tqdm
+    
+class CLIPModelWrapper:
+    
+    def __init__(
+        self,
+        model_name: str,
+        device: str = "cuda" if torch.cuda.is_available() else "cpu",
+        **kwargs: Any,
+    ):
+        self.model_name = model_name
+        self.device = device
+        self.model = AutoModel.from_pretrained(model_name).to(self.device)
+        self.processor = AutoProcessor.from_pretrained(model_name)
+
+    def preprocess(
+            self,
+            texts: list[str],
+            images: list[Image.Image],
+    ):
+        return self.processor(text=texts, images=images, return_tensors="pt", padding=True)
+
+    def get_text_embeddings(self, texts: list[str], batch_size: int = 32):
```

Comment:
is there any generality to these? The text encoder is built around the SentenceTransformers object. It might be nice to do the same here (use some sort of semi-standard interface so that we don't need to wrap everything)